lib/auth/auth_with_roles.go

```diff
@@ -1041,41 +1034,38 @@ func (a *AuthWithRoles) GenerateUserCerts(ctx context.Context, req proto.UserCer
 	}, nil
 }
 
-func (a *AuthWithRoles) CreateSignupToken(user services.UserV1, ttl time.Duration) (token string, e error) {
-	if err := a.action(defaults.Namespace, services.KindUser, services.VerbCreate); err != nil {
-		return "", trace.Wrap(err)
-	}
-	return a.authServer.CreateSignupToken(user, ttl)
-}
-
-func (a *AuthWithRoles) GetSignupTokenData(token string) (user string, otpQRCode []byte, err error) {
+func (a *AuthWithRoles) GetSignupU2FRegisterRequest(token string) (u2fRegisterRequest *u2f.RegisterRequest, e error) {
 	// signup token are their own authz resource
-	return a.authServer.GetSignupTokenData(token)
+	return a.authServer.CreateSignupU2FRegisterRequest(token)
 }
 
-func (a *AuthWithRoles) GetSignupToken(token string) (*services.SignupToken, error) {
-	// signup token are their own authz resource
-	return a.authServer.GetSignupToken(token)
-}
+func (a *AuthWithRoles) CreateUserToken(req CreateUserTokenRequest) (services.UserToken, error) {
+	if err := a.action(defaults.Namespace, services.KindUser, services.VerbUpdate); err != nil {
```

Comment:
Why was this changed from `services.VerbCreate` to `services.VerbUpdate`?

Reply:
This strikes me as a naming issue; I think `CreateUserToken` is creating a *password reset* token, which does effectively map to `VerbUpdate` with our current model.

Reply:
We no longer create a user when processing signup token. `UserToken` is about resetting a password and generating a secure token to set up a new one. Which is essentially user update operation.

Reply:
Does it make sense to rename `CreateUserToken` to `CreatePasswordResetToken` and consequanlty `UserToken` to `PasswordResetToken` ? The former gives a little bit room for other use cases moving forward as far as naming is concern.

Reply:
Renamed to ResetPasswordToken